internal/plugin/connectors/http/generic/connector.go

```diff
@@ -14,6 +15,40 @@ type Connector struct {
 	config *config
 }
 
+func addOAuth1Header(c *Connector, credentialsByID connector.CredentialValuesByID, r *gohttp.Request) error {
+	oAuth1Params, _ := renderTemplates(c.config.OAuth1Secrets, credentialsByID)
```

Comment:
I know we did this and tested the length of `oAuth1Params` for a reason, I'm just having troubling remembering exactly why(Which is exactly why we should've made a comment).I'm 99% sure it's because renderTemplates will throw an err if `oauth1` is not declared in the config. So we check the length of the returned map instead. I'm going to double check this and make a comment explaining why

Reply:
This came from a comment by @sgnn7 [here](https://github.com/cyberark/secretless-broker/pull/1321#discussion_r465341047.

We only want to run our code if the rendered template comes back with a length > 0. If we didn't do this, we'd have to declare `oauth1` in the config every time. Do you think this is the right solution to this problem?

cc: @mdodell

Reply:
We have error handling for someone declaring `oauth1` in the config, but not providing the necessary values [here](https://github.com/cyberark/secretless-broker/blob/deda81e5e06862d8dbbfdca06eea298c6da42b0f/internal/plugin/connectors/http/generic/oauth/v1/protocol.go#L88)

Reply:
My concern is if there is an issue rendering the templates, before we check what the length is, because the rendering itself may have failed. We don't need to call `addOAuth1Header` every time, but if we do, we should confirm that there were no errors when calling `renderTemplates` inside of it. 

`renderTemplates` shouldn't throw an error if `c.config.OAuth1Secrets` is empty. Further, `c.config.OAuth1Secrets` will never be `nil` because `1stringsToTemplates` sets it to an empty map if no values are given.

@sgnn7 what do you think?